frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.tsx

```diff
@@ -147,7 +151,13 @@ const DatabaseFormBody = ({
   setIsDirty,
   config,
 }: DatabaseFormBodyProps): JSX.Element => {
-  const { values, dirty } = useFormikContext<DatabaseData>();
+  const { values, dirty, setFieldValue, validateForm } =
+    useFormikContext<DatabaseData>();
+  const [pendingConnectionString, setPendingConnectionString] = useState<
+    string | null
+  >(null);
+  const [needsValidationAfterPaste, setNeedsValidationAfterPaste] =
```

Comment:
Because dirty state is stubborn and require a cycle after validating the form programmatically to enable the Save button